Zend/zend_execute.c

```diff
@@ -3559,21 +3560,21 @@ static zend_never_inline zend_result zend_fetch_static_property_address_ex(zval
 			ce = zend_fetch_class(NULL, opline->op2.num);
 			if (UNEXPECTED(ce == NULL)) {
 				FREE_OP(op1_type, opline->op1.var);
-				return FAILURE;
+				return NULL;
 			}
 		} else {
 			ce = Z_CE_P(EX_VAR(opline->op2.var));
 		}
-		if (EXPECTED(op1_type == IS_CONST) && EXPECTED(CACHED_PTR(cache_slot) == ce)) {
-			*retval = CACHED_PTR(cache_slot + sizeof(void *));
+		if (EXPECTED(op1_type == IS_CONST) && EXPECTED(CACHED_PTR(cache_slot + sizeof(void *)) == ce)) {
```

Comment:
It doesn't seem right that you add `sizeof(void *)` here. On line 3555 you store `ce` in `cache_slot` not in `cache_slot + sizeof(void *)`; and `CACHE_POLYMORPHIC_PTR` also takes offset 0 to the `cache_slot` for ce.

Reply:
> It doesn't seem right that you add `sizeof(void *)` here. `CACHE_POLYMORPHIC_PTR` takes offset 0 to the `cache_slot` for ce. The `result` is stored in offset `sizeof(void *)` instead of `ce`.

You are right! Thanks!